scraibe/diarisation.py

```diff
@@ -210,53 +211,64 @@ def load_model(cls,
-            
-        elif os.path.exists(model) and not use_auth_token:
-            # check if model can be found locally nearby the config file
-            with open(model, 'r') as file:
-                config = yaml.safe_load(file)
-            
-            path_to_model = config['pipeline']['params']['segmentation']
-
-            if not os.path.exists(path_to_model):
-                warnings.warn(f"Model not found at {path_to_model}. "\
-                    "Trying to find it nearby the config file.")
+            if cache_token and use_auth_token is not None:
+                cls._save_token(use_auth_token)
+                
+            if not os.path.exists(model) and use_auth_token is None:
+                use_auth_token = cls._get_token()
                 
-                pwd = model.split("/")[:-1]
-                pwd = "/".join(pwd)
+            elif os.path.exists(model) and not use_auth_token:
+                # check if model can be found locally nearby the config file
+                with open(model, 'r') as file:
+                    config = yaml.safe_load(file)
                 
-                path_to_model = os.path.join(pwd, "pytorch_model.bin")
+                path_to_model = config['pipeline']['params']['segmentation']
 
                 if not os.path.exists(path_to_model):
-                    warnings.warn(f"Model not found at {path_to_model}. \
-                        'Trying to find it nearby .bin files instead.")
-                    # list elementes with the ending .bin
-                    bin_files = [f for f in os.listdir(pwd) if f.endswith(".bin")]
-                    if len(bin_files) == 1:
-                        path_to_model = os.path.join(pwd, bin_files[0])
-                    else:
-                        warnings.warn("Found more than one .bin file. "\
-                            "or none. Please specify the path to the model " \
-                            "or setup a huggingface token.")
-                                            
-                warnings.warn(f"Found model at {path_to_model} overwriting config file.")
-                
-                config['pipeline']['params']['segmentation'] = path_to_model
-                
-                with open(model, 'w') as file:
-                    yaml.dump(config, file)
-                              
-        _model =  Pipeline.from_pretrained(model,
-                                           use_auth_token = use_auth_token,
-                                           cache_dir = cache_dir,
-                                           hparams_file = hparams_file,)
+                    warnings.warn(f"Model not found at {path_to_model}. "\
+                        "Trying to find it nearby the config file.")
+                    
+                    pwd = model.split("/")[:-1]
+                    pwd = "/".join(pwd)
+                    
+                    path_to_model = os.path.join(pwd, "pytorch_model.bin")
```

Comment:
Why only move it here? And maybe we could find a more novel approach in the future. Perhaps do a TODO